packages/shared/src/variants/baduk.ts

```diff
@@ -175,6 +197,10 @@ export class Baduk extends AbstractGame<NewBadukConfig, BadukState> {
     });
   }
 
+  /**
+   * Called after placing stone and resolving captures. Validates if move is allowed by checking if it results in self-capture or a previously seen position.
```

Comment:
These "Baduk lifecycle" methods especially benefit from the documentation, since many variants override them. 👍

Might be good to document what is expected of this function - specifically, that it should throw on an invalid move.